payjoin/tests/integration.rs

```diff
@@ -985,11 +987,168 @@ mod integration {
+                    panic!("proposal should exist");
+                };
+                let (payjoin_proposal, fallback_tx) =
+                    handle_directory_proposal(&receiver, proposal, None)?;
+
+                let (req, ctx) =
+                    payjoin_proposal.create_post_request(services.ohttp_relay_url().as_str())?;
+                let response = agent
+                    .post(req.url)
+                    .header("Content-Type", req.content_type)
+                    .body(req.body)
+                    .send()
+                    .await?;
+                let monitoring_payment = payjoin_proposal
+                    .process_response(&response.bytes().await?, ctx)
+                    .save(&recv_persister)?;
+
+                // **********************
+                // Inside the Sender:
+                // Sender does not do anything with the Payjoin proposal. Instead, they just
+                // broadcast the fallback transaction.
+                // let payjoin_tx = extract_pj_tx(&sender, fallback_tx)?;
+                sender.send_raw_transaction(&fallback_tx)?;
+                tracing::info!("original proposal (fallback transaction) sent");
+
+                // **********************
+                // Inside the Receiver:
+                // monitor the payment on the receiver side
+                monitoring_payment.check_payment(
+                    |txid| {
+                        let get_tx_result = receiver.get_raw_transaction(txid);
+                        match get_tx_result {
+                            Ok(tx) => {
+                                Ok(Some(tx.transaction().expect("transaction should be decodable")))
+                            }
+                            Err(_) => Ok(None)
+                        }
+                    },
+                    |_| {
+                        panic!("This function should never be invoked as the fallback transaction check happens before in the monitor function")
+                    },
+                )
+                    .save(&recv_persister)
+                    .expect("receiver should successfully monitor for the payment");
+
+                assert_eq!(
+                recv_persister.load().unwrap().last(),
+                Some(payjoin::receive::v2::SessionEvent::Closed(
+                    payjoin::receive::v2::SessionOutcome::FallbackBroadcasted
+                )),
+                "The last event of the persister should show that the receiver detected sender broadcasting the fallback transaction",
+            );
+
+                Ok(())
+            }
+
+            Ok(())
+        }
+
         fn handle_directory_proposal(
             receiver: &corepc_node::Client,
             proposal: Receiver<UncheckedOriginalPayload>,
             custom_inputs: Option<Vec<InputPair>>,
-        ) -> Result<Receiver<PayjoinProposal>, BoxError> {
+        ) -> Result<(Receiver<PayjoinProposal>, Transaction), BoxError> {
```

Comment:
IIRC `build_sweep_psbt` should return a signed PSBT. In which case we don't need to change this method. The sender can just decide to fallback using the PSBT they already signed. 
Alternatively you can access the fallback via the session history https://github.com/payjoin/rust-payjoin/blob/master/payjoin/src/core/send/v2/session.rs#L59.

Eitherway I don't think this we need to touch this method.